### PR TITLE
gh-100450: Add `sqlite.Row.__contains__` method

### PR DESCRIPTION
**Pull request by sobolevn**
I went ahead and added it :)

CC @erlend-aasland and @vsajip 

* Issue: gh-100450

### CODE REVIEW

File: Modules/_sqlite/row.c [hunk 1]
```diff
@@ -96,6 +96,25 @@ PyObject* pysqlite_row_item(pysqlite_Row* self, Py_ssize_t idx)
    return Py_XNewRef(item);
 }
 
+static int
+pysqlite_row_contains(pysqlite_Row* self, PyObject *arg)
+{
+    Py_ssize_t nitems, i;
```
Comment (erlend-aasland) [inline Modules/_sqlite/row.c]: Please use intermingled declarations :)

File: Modules/_sqlite/row.c [hunk 2]
```diff
@@ -96,6 +96,25 @@ PyObject* pysqlite_row_item(pysqlite_Row* self, Py_ssize_t idx)
...
+    int cmp = 0;
+
+    nitems = PyTuple_Size(self->description);
+
+    for (i = 0; cmp == 0 && i < nitems; i++) {
+        cmp = PyObject_RichCompareBool(
+            arg,
+            PyTuple_GET_ITEM(PyTuple_GET_ITEM(self->description, i), 0),
```
Comment (erlend-aasland) [inline Modules/_sqlite/row.c]: There's a lot happening on this line. I'd prefer it if you extracted these items before calling `PyObject_RichCompareBool`, for improved readability.

**Review by erlend-aasland — CHANGES_REQUESTED** [review]
I left some nitpicks. Also, please add tests.

**Comment by pochmann** [comment]
@erlend-aasland Hmm, is this whole thing not wrong anyway? (See my comment in the issue thread.)

**Comment by sobolevn** [comment]
@erlend-aasland 
> Also, please add tests.

I've added two doctests, aren't they enough?
If not, I don't know what is the best place for these tests. I can make new file for `Row` tests 🤔

**Comment by erlend-aasland** [comment]
> I've added two doctests, aren't they enough? If not, I don't know what is the best place for these tests.

The doctests are there to make sure the examples in the docs actually work; it is not the place for unit tests. The unit tests (and regression tests, functional tests, etc.) all live in `Lib/test/test_sqlite3`. Their purpose is to test the `_sqlite` extension module (and the `sqlite3` module) in all possible ways; we aim for as high code coverage as possible.

If I want to check the code coverage of the `_sqlite` C code, I can do so easily with a couple of compiler/linker flags, running `./python.exe -m test test_sqlite3`, and generate a report.

Also, I'm not sure the buildbots run doctests; IIRC, they only run the test suite, but I might be wrong about this.

**Comment by sobolevn** [comment]
Sure, @erlend-aasland, I've added a unit test :)

**Comment by erlend-aasland** [comment]
> Sure, @erlend-aasland, I've added a unit test :)

Thanks! BTW; I haven't decided if this is a good thing to add, or not. But lets keep that conversation on the issue. I'm going to mark this with do-not-merge until the discussion has landed.

File: Lib/test/test_sqlite3/test_factory.py [hunk 3]
```diff
@@ -240,6 +240,14 @@ def test_sqlite_row_as_sequence(self):
         self.assertEqual(list(reversed(row)), list(reversed(as_tuple)))
         self.assertIsInstance(row, Sequence)
 
+    def test_sqlite_row_contains(self):
+        """Checks if the row object can be used with `in`"""
```
Comment (erlend-aasland) [inline Lib/test/test_sqlite3/test_factory.py:244]: FYI, these kind of comments result in weird `unittest` output in verbose mode; it is better to use verbose test names :)

Comment (sobolevn) [reply Lib/test/test_sqlite3/test_factory.py:244]: TIL! I was falling the style of other test cases in this file.

Comment (erlend-aasland) [reply Lib/test/test_sqlite3/test_factory.py:244]: I misremembered! :) It is _docstrings_ that must be avoided. Ordinary comments are quite ok.

**Comment by vsajip** [comment]
> I haven't decided if this is a good thing to add

Should we open this to wider discussion on Discourse/Core development?

**Comment by erlend-aasland** [comment]
Yeah, IMO it is worth it trying to gather more feedback on Discourse.

**Pull request closed without merging**